src/v/kafka/server/handlers/fetch.cc

```diff
@@ -331,7 +331,30 @@ static ss::future<std::vector<read_result>> fetch_ntps_in_parallel(
   std::vector<ntp_fetch_config> ntp_fetch_configs,
   bool foreign_read,
   std::optional<model::timeout_clock::time_point> deadline) {
-    return ssx::parallel_transform(
+    size_t total_max_bytes = 0;
+    size_t total_max_bytes_strict = 0;
+    for (const auto& c : ntp_fetch_configs) {
+        total_max_bytes += c.cfg.max_bytes;
+        total_max_bytes_strict += c.cfg.strict_max_bytes;
+    }
+
+    auto max_bytes_per_fetch
+      = config::shard_local_cfg().kafka_max_bytes_per_fetch();
+    if (total_max_bytes > max_bytes_per_fetch) {
+        auto per_partition = max_bytes_per_fetch / ntp_fetch_configs.size();
```

Comment:
if `max_bytes_per_fetch` is 1, and `npts.size() == 2` you get 0 bytes per partition and an infinite loop no? 

you prolly need a `max(per_partition, 1mb)` or smth like that as a safe default